vllm/model_executor/models/whisper.py

```diff
@@ -0,0 +1,445 @@
+                raise ValueError(
+                    "Encoder self-attention step. The KV cache should not be populated."
+                )
+            k, _ = self.k_proj(hidden_states)
+            v, _ = self.v_proj(hidden_states)
+
+            # reshape the tensors to the shape required by the EncoderAttention
+            proj_shape = (bsz, -1, self.head_dim * self.num_heads)
+            q = q.reshape(*proj_shape)
+            k = k.reshape(*proj_shape)
+            v = v.reshape(*proj_shape)
+            input_metadata.attn_bias = None
+            attn_output = self.attn(q, k, v, input_metadata)
+
+        o, _ = self.out_proj(attn_output)
+
+        return o
+
+
+class WhisperEncoderBlock(nn.Module):
+
+    def __init__(self,
+                 config: WhisperConfig,
+                 linear_method: Optional[LinearMethodBase] = None):
+        super().__init__()
+        self.d_model = config.d_model
+
+        self.self_attn = WhisperAttention(
+            config=config,
+            num_heads=config.encoder_attention_heads,
+            linear_method=linear_method,
+        )
+        self.self_attn_layer_norm = nn.LayerNorm(self.d_model)
+        self.activation_fn = GELUActivation()
+        self.fc1 = ColumnParallelLinear(self.d_model, config.encoder_ffn_dim)
+        self.fc2 = RowParallelLinear(config.encoder_ffn_dim, self.d_model)
+        self.final_layer_norm = nn.LayerNorm(self.d_model)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        input_metadata: InputMetadata,
+    ) -> torch.Tensor:
+
+        residual = hidden_states
+
+        hidden_states = self.self_attn_layer_norm(hidden_states)
+        hidden_states = self.self_attn(hidden_states, None, input_metadata)
+        hidden_states = residual + hidden_states
+
+        residual = hidden_states
+
+        hidden_states = self.final_layer_norm(hidden_states)
+        hidden_states, _ = self.fc1(hidden_states)
+        hidden_states = self.activation_fn(hidden_states)
+        hidden_states, _ = self.fc2(hidden_states)
+
+        hidden_states = residual + hidden_states
+
+        return hidden_states
+
+
+class WhisperEncoder(nn.Module):
+
+    def __init__(self,
+                 config: WhisperConfig,
+                 linear_method: Optional[LinearMethodBase] = None):
+        super().__init__()
+        self.d_model = config.d_model
+        self.num_mel_bins = config.num_mel_bins
+        self.max_source_positions = config.max_source_positions
+
+        self.conv1 = nn.Conv1d(self.num_mel_bins,
+                               self.d_model,
+                               kernel_size=3,
+                               padding=1)
+        self.conv2 = nn.Conv1d(self.d_model,
+                               self.d_model,
+                               kernel_size=3,
+                               stride=2,
+                               padding=1)
+
+        self.embed_positions = nn.Embedding(self.max_source_positions,
```

Comment:
could this be `VocabParallelEmbedding`?

Reply:
yes, this part is definitely parallelizable, my omission, I did introduce the `VocabParallelEmbedding` here before 🥴